sdk/src/wallet/operations/transaction/high_level/delegation/delay.rs

```diff
@@ -0,0 +1,134 @@
+// Copyright 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    client::{api::PreparedTransactionData, secret::SecretManage},
+    types::block::{
+        context_input::{CommitmentContextInput, ContextInput},
+        output::{AddressUnlockCondition, BasicOutput, DelegationId, DelegationOutputBuilder, MinimumOutputAmount},
+    },
+    wallet::{operations::transaction::TransactionOptions, types::TransactionWithMetadata, Wallet},
+};
+
+impl<S: 'static + SecretManage> Wallet<S>
+where
+    crate::wallet::Error: From<S::Error>,
+    crate::client::Error: From<S::Error>,
+{
+    pub async fn delay_delegation_claiming(
+        &self,
+        delegation_id: DelegationId,
+        reclaim_excess: bool,
+    ) -> crate::wallet::Result<TransactionWithMetadata> {
+        let prepared_transaction = self
+            .prepare_delay_delegation_claiming(delegation_id, reclaim_excess)
+            .await?;
+
+        self.sign_and_submit_transaction(prepared_transaction, None, None).await
+    }
+
+    pub async fn prepare_delay_delegation_claiming(
```

Comment:
Please add some doc comments that explain what this is doing with the reclaim_excess bool

Reply:
done